### PR TITLE
DM-30927: Set calexp WCS to None for failed astrometric fit

### PR DESCRIPTION
**Pull request by laurenam**

### CODE REVIEW

File: python/lsst/faro/measurement/DetectorMeasurement.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             else:
                 externalPhotoCalibCatalog = inputs.pop("externalPhotoCalibTractCatalog")
             row = externalPhotoCalibCatalog.find(detector)
-            externalPhotoCalib = row.getPhotoCalib()
+            externalPhotoCalib = None if row is None else row.getPhotoCalib()
```
Comment (erykoff) [inline python/lsst/faro/measurement/DetectorMeasurement.py:162]: How has this not been failing already?

File: python/lsst/faro/measurement/TractMeasurement.py [hunk 2]
```diff
@@ -113,6 +113,14 @@ def run(
     ):
         data = defaultdict(list)
         for catalog, photoCalib, astromCalib, dataId in zip(catalogs, photoCalibs, astromCalibs, dataIds):
+            if self.config.requireAstrometry and astromCalib is None:
+                self.log.info("requireAtrometry is True but astromCalib is None for %s.  Skipping...",
```
Comment (erykoff) [inline python/lsst/faro/measurement/TractMeasurement.py:117]: How common are these logs going to be?

Also, `requireAtrometry` is missing an s.

**Review by erykoff — APPROVED** [review]
A couple of minor comments.

File: python/lsst/faro/measurement/TractMeasurement.py (hunk 2 above)
Comment (laurenam) [reply python/lsst/faro/measurement/TractMeasurement.py:117]: I'm assuming pretty rare (or we've got bigger problems!)

Typo fixed...thanks for another good catch!

File: python/lsst/faro/measurement/DetectorMeasurement.py (hunk 1 above)
Comment (laurenam) [reply python/lsst/faro/measurement/DetectorMeasurement.py:162]: 🤷

**Pull request merged**